Src/modules/dronecan/core/dronecan_module.cpp

```diff
@@ -33,6 +33,11 @@ void DronecanModule::init() {
 }
 
 void DronecanModule::spin_once() {
+    if (uavcanCheckProtocol() == CanardFrameProtocol::CANARD_FRAME_CYPHAL) {
```

Comment:
Consider modifying ModuleManager::get_active_protocol() instead. It is better to perform protocol check once on the initialization stage rather that check it in each spin call of each module.